tests/apps/test_similarity.py

```diff
@@ -444,18 +481,29 @@ def test_mean_computation_event(self, monkeypatch):
         6 photons, and max 3 photons per mode."""
         graph = nx.complete_graph(6)
         with monkeypatch.context() as m:
-            m.setattr(
-                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
-                lambda *args, **kwargs: 1.0 / 336,
-            )
-
+            m.setattr("strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                      lambda *args, **kwargs: 1.0 / 336)
             assert np.allclose(similarity.prob_event_mc(graph, 6, 3), 1.0)
 
+    @pytest.mark.parametrize("k", [3, 5, 7, 9])
+    @pytest.mark.parametrize("nmax", [1, 2])
+    def test_odd_photon_numbers(self, k, nmax):
+        """Test if function returns zero probability for odd number of total photons."""
+        graph = nx.complete_graph(10)
+        assert similarity.prob_event_mc(graph, k, nmax) == 0.0
+
     def test_max_loss(self):
         """Test if function samples from the vacuum when maximum loss is applied."""
         graph = nx.complete_graph(6)
         assert similarity.prob_event_mc(graph, 6, 3, samples=1, loss=1) == 0.0
 
+    def test_known_result(self):
```

Comment:
Same comment as before

Reply:
It is hard to encode a test for monte carlo functions as outputs change on every run.